src/controller.cpp

```diff
@@ -659,7 +659,12 @@ void Controller::connected(const QString& pubkey,
   // We have succesfully completed all pending connections.
   logger.debug() << "Connected from state:" << m_state;
   setState(StateOn);
-  emit newConnectionSucceeded();
+
+  logger.debug() << "Connection happened due to server switch:"
+                 << isActivatingFromSwitch;
```

Comment:
This debug statement threw me off for a second because the text initially made me assume that we've encountered a server switch but then realized that depends on the value of `isActivatingFromSwitch`. For the sake of clarity can this be changed to be more explicit in when we log the statement? Possibly:

```suggestion
  if (isActivatingFromSwitch)
  {
     logger.debug() << "Connection happened due to server switch. Do not collect telemetry regarding new connection.";
  }
```

Reply:
Added something similar, with logging for both ways. Good call on the confusion there.